### ISSUE TITLE
Subsonic (Airsonic) server remote issue

### ISSUE THREAD

**Issue by gerroon** [ampache/ampache#1621]
Hi

I am trying to add a working Airsonic instance to Ampache and I am getting errors (over and over) like below after adding the remote catalog.

"Server sent events connection error. Re-connection"

Airsonic is running on http://127.0.0.1:8080/airsonic so this is the url I put in the field along with my airsonic creds.

There is nothing printing in the log at all regarding the issue, in fact no lines had been printed since adding the remote repo

I am not sure if there is an issue with api or a hiding bug.

I am using
ampache 3.8.4 develop 
Debian 7.1 x32
ffmpeg 1.0.10
Apache 2.2.22
Php 5.6.31
Mysql 14.14

My url is
http://xx.xx.xx.xx/ampache-dev

Airsonic
http://127.0.0.1:8080/airsonic

CONFIG
https://paste.debian.net/hidden/ec5a8be2/

PHP
https://paste.debian.net/hidden/08861a8e/

AIRSONIC
https://github.com/airsonic/airsonic

**Comment by wagnered** [comment]
@gerroon,

I'm not sure what you are trying to do with this setup.

Please read [https://github.com/ampache/ampache/wiki/API](https://github.com/ampache/ampache/wiki/API) for information about Ampache's API and accessing music via Subsonic API and how to set it up for accessing music from Ampache multimedia server.

**Comment by gerroon** [comment]
@wagnered 

I am adding my subsonic server as a remote catalog which is listed in the catalog type when adding a catalog.

I have a working subsonic api with my subsonic client (Dsub on Android) if you mean running Ampache as a Subsonic impersonator.

**Comment by gerroon** [comment]
"Starting with 3.7.0, Ampache now support more Remote catalog type in addition to XML-RPC. You can use a SubSonic server, a Dropbox folder or your SoundCloud account as a remote catalog for Ampache."

**Comment by wagnered** [comment]
Sorry, misread your first message.  Understand now and will look into
the issue.

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by gerroon** [comment]
I am sorry if I did not articulate the issue in the first place. Thanks for considering.

**Comment by wagnered** [comment]
I just installed airsonic, created a subsonic catalog and was able to play songs via ampache's web player.

Make sure that "Use SubSonic backend" is enabled  on the admin->system page.

I also have "url rewrite" enabled on the Admin->streaming page, but I haven't checked if that is actually required.

When I first tried it, I got ampache error "Artist: ERROR" but that was because I had used the default password, but already had reset the airsonic password.

At any rate, It must be in your admin/config settings.

**Comment by gerroon** [comment]
@wagnered 
Thanks for looking into it. I will try these now. I have the backend enabled already let me see what is going on.

**Comment by gerroon** [comment]
Interesting, first thing first. I do not have url rewrite in streaming. I only have "use beautiful stream url". Is that it?

**Comment by wagnered** [comment]
In the Ampache admin->streaming settings page "Enable url rewriting" is the first listing on the page.

**Comment by gerroon** [comment]
yeah I am not having any luck with it. I did all the checks based on your recommendations, all are good. I still cant find the url rewrite option. I assume you meant beautiful urls which is enabled.

"Server-Sent Events connection error. Re-connection.."

**Comment by gerroon** [comment]
@wagnered  

I do not have that setting

http://pix.toile-libre.org/upload/original/1508039669.png

**Comment by wagnered** [comment]
The image is displaying the "Preferences" page, not the admin tab ->server-config->streaming page.

**Comment by gerroon** [comment]
@wagnered 

Man I feel like a  dumb duck here but it is the same page see your self and still missing the url

http://pix.toile-libre.org/upload/original/1508040641.png

**Comment by gerroon** [comment]
I even did "write a new config" under debug still no such thing showed up after restarting the server.

**Comment by wagnered** [comment]
What version ampache is installed, your images don't display it.

Here is what should be displayed
![screenshot at 2017-10-14 21-17-08](https://user-images.githubusercontent.com/6115438/31581656-98cb395c-b125-11e7-878d-7924ea4bd9e4.png)
Maybe it's aproblem with one of your settings.

**Comment by gerroon** [comment]
I am using 3.8.4 develop

**Comment by gerroon** [comment]
Yeah It looks like I am missing that setting. I have no idea about which setting might cause it. Maybe a database migration in the past messed it up? 

If this is saved in the database maybe I can try editing it? If so which database field?

**Comment by gerroon** [comment]
here is the actual full screenshot

actual full screenshot

http://pix.toile-libre.org/upload/original/1508042526.png

**Comment by wagnered** [comment]
Did you install 3.8.4-develop and not overwrite the database? That is not good as fields were added/removed since 3.8.3-develop. Actually, from the image, it might even be from a much earlier version.

I suggest you reinstall 3.8.4-develop and overwrite the existing DB.  For now you can try setting `ampache`.'preferences` with name - "stream_beautiful_url" to value = 1 using mysql command line.

**Comment by gerroon** [comment]
What do you mean not overwrite the database? I clone the repo, installed composer and copied my config and edited the rest related files for correct url. This is how the upgrade co recommended.

**Comment by gerroon** [comment]
Btw stream beautiful url=1 still did not help with airsonic database. I think I will have to do a fresh install based on your input.

**Comment by wagnered** [comment]
Yea That might have been ok before major changes to the database began.

If new tables/columns are added, it is necessary to do a fresh install by moving/renaming ampache.cfg.php to another name. this will cause Ampache to start a new install. After the install, mv the config file back to ampache.cfg.php.  If there are new tables/columns to be added, it is necessary to overwrite the existing database.

But even this is not always the correct procedure, because sometimes the config file itself is  updated to a new version.  The config file has a line: "config_version = 34" which is the current version, and ampache checks and makes sure the config version matches what ampache is looking for.

**Comment by gerroon** [comment]
Thanks for the insight. Do have I have to wipe my database as well?

**Comment by wagnered** [comment]
I think it would be best to overwrite the database during a fresh install. it seems to be the only way to get any new tables/columns inserted/updated.

**Comment by gerroon** [comment]
Ok
I did a full fresh reinstall with a new config and empty database. I now have the url rewrite. But that did not solve the issue of "Server-Sent Events connection error. Re-connection.." with an Airsonic repo.

What is the exact url scheme do you use for airsonic? My airsonic is not sitting at the top of the web server. I also use Tomcat8 to serve it not sure :(

**Comment by wagnered** [comment]
I used the standalone version and set it up as described in the
instructions.  The command I used was: `java -jar -Dserver.port=8100
airsonic.war` because something else was using port 8080.

Also, I have mod_rewrite enabled for my Apache server.

Are you able to create the catalog?


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by gerroon** [comment]
I mean the exact url you put for the subsonic catalog. 

I use http://xx.xx.xx.xx:8080/airsonic/  in the url field of the remote subsonic repo.

I have mod_rewrite for Apache as well. I was able to use Dsub client to connect to Ampache's Subsonic Api, so all should be working no problem.

I still get the same "Server-Sent Events connection error. Re-connection.." when adding a Subsonic remote catalog.

**Comment by wagnered** [comment]
That would be http://127.0.0.1:8100. The airsonic is running from /var/airsonic which would make it DOCROOT..  Yes, everything is set up  to operate on one computer which limits me to "localhost".

And the ampache server is http://127.0.0.1:80.

**Comment by gerroon** [comment]
Well my server is not at the root. And that is the issue. It sounds like AMpache expects root level url.

I just reverse proxied my running airssonic url to http://xx.xx.xx.xx:880  and now the catalog update seems to be happening without error, at least I am not getting that server connection anymore.

So I am not sure if you treat this as a bug, or a feature or a non issue. But I would say that not everyone can install or proxy their ***sonics to the root of the server.

**Comment by wagnered** [comment]
Is xx.xx.xx.xx (airsonic) on the same network or external to the network your ampache server is on?  

With all those x's I can't make any sense of the relationship between the url's of the  servers, excluding the reverse  proxy, so I cannot respond.

**Comment by gerroon** [comment]
Well 

http://192.168.2.100:8080/airsonic/ (is the actual url)

http://192.168.2.100:880 (is the reverse proxied to simulate root level access)

**Comment by wagnered** [comment]
You first indicated that the airsonic server was on localhost.

I just started my airsonic server with the address http://192.168.2.3:8100, deleted the airsonic catalog and recreated it with the URI of http://192.168.2.3:8100.

Also you might want to check the config file for web_path and set it according to the comment.

I'm not convinced that it is necessary to run a reverse proxy to access a server on the same network.

**Comment by wagnered** [comment]
Actually about my last message: I logged into my ampache server with the URI of http://192.168.2.1:80 so it was on the same network as the airsonic server.

The Apache web server is set up to watch only port 80, it automatically watches all net interfaces.

**Comment by gerroon** [comment]
Well local host meaning that itboth ampache and airsonic are running on the same server.

I reverse proxied because I cant run airsonic as the web root, tomcat runs airsonic as a sub url.  I also do not want airsonic to take over my server root in the first place. 

What I am saying is that Ampache does not like "http://192.168.2.100:8080/airsonic/ ". Try it yourself with Tomcat server (Install tomcat and put airsonic under webapps folder of tomcat). You will see that Ampache cant deal with that url.

However Ampache is fine with "http://192.168.2.100:881" or whatever your airsonic running at doc root. You were able to get airsonic to be runnic at that root level because you ran it as an individual app. If you run via Tomcat you wont get root level access.

**Comment by gerroon** [comment]
I guess, to summerize it, current Ampache Subsonic backend expects put http://IP:PORT type Subsonic url, it does not want to see antyhing beyond that PORT part. At least that was my experience in last couple hours. 

Once I got that kind of url goung (via reverse proxying it) it worked.

**Comment by wagnered** [comment]
>Well local host meaning that itboth ampache and airsonic are running on the same server.

NO localhost means that it is working disconnected from the network. My example above was with two different servers(apache and airsonc standalone) on the same computer using the same local network (192.168.2.3) with different ports.

I thought the standalone server had the tomcat server built in. but I now cannot find any reference to that assumption.

When I get some more time I'll explore it further

Anyway, It's getting late. so I'm shutting down for now.

**Comment by gerroon** [comment]
Airsonic uses embedded Tomcat when runs as a standalone app. You are right about it. 

To cap it, the issue was not that Ampache and Subsonic were on a different network or on a localhost or connecting over the internet. The ip or the url name does not cause the network connection issue with Subsonic remote catalog in Ampache. Trust me I tried localhost, and couple other ways to access to Subsonic and none worked until I was able to turn the SUbsonic url into http:/IP:PORT 

Basically 

**Subsonic server urls that work**
http://IP:PORT
http://URL:PORT


**Subsonic server urls that does not work**
(my server setup requires the second slash)
http://IP:PORT/Airsonic/
http://URL:PORT/Airsonic/
http://IP:PORT/Airsonic
http://URL:PORT/Airsonic 


Anyway thanks for your help. WIthout your input I could not have figure out the actual issue.

**Comment by wagnered** [comment]
@gerroon,

Well, as I cannot duplicate your issue, I will wait until someone else
reports the same problem.

You are welcome.

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug
